Sources/CreateAPI/Generator/Templates.swift

```diff
@@ -343,10 +346,23 @@ final class Templates {
         """
     }    
     
-    func encodeOneOf(properties: [Property]) -> String {
-        let statements = properties.map {
-            "case .\($0.name)(let value): try container.encode(value)"
+    func encodeOneOf(properties: [Property], discriminator: Discriminator?) -> String {
+        var statements: [String] = []
+        if let discriminator = discriminator {
+            for property in properties {
+                let correspondingMappings = discriminator.mapping.filter { $1 == property.type }.sorted { $0.key < $1.key }
+                if !correspondingMappings.isEmpty {
+                    for mapping in correspondingMappings {
+                        statements.append("case .\(mapping.key)(let value): try container.encode(value)")
+                    }
+                }
```

Comment:
Passing an empty collection to a `for` loop isn't an issue so we can avoid the `!isEmpty` check here

```suggestion
                for mapping in correspondingMappings {
                    statements.append("case .\(mapping.key)(let value): try container.encode(value)")
                }
```